src/OrchardCore.Modules/OrchardCore.Media/Services/MediaFileStoreResolverMiddleware.cs

```diff
@@ -79,7 +79,7 @@ public async Task Invoke(HttpContext context)
 
             // When multiple requests occur for the same file we use a Lazy<Task>
             // to initialize the file store request once.
-            await _workers.GetOrAdd(subPathValue, x => new Lazy<Task>(async () =>
+            await _workers.GetOrAdd(subPathValue, subPathValue => new Lazy<Task>(async () =>
```

Comment:
Yes and no, still the same problem if used like this

            await _workers.GetOrAdd(
                subPathValue,
                x => new Lazy<Task>(SomeMethodAsync(subPathValue))).Value;

In place of being used like this

            await _workers.GetOrAdd(
                subPathValue,
                x => new Lazy<Task>(SomeMethodAsync(x))).Value;

Or like that

            await _workers.GetOrAdd(
                subPathValue,
                subPathValue => new Lazy<Task>(SomeMethodAsync(subPathValue))).Value;

Would need more time to simulate it because we need concurrent requests but with different media paths, so not so obvious without using a testing app to simulate these requests.

In the meantime, maybe worth to first try the simple suggested change before trying another fix if needed, like using a simple async semaphore.

Reply:
Thanks, I'll try, though I couldn't reproduce the issue locally.

Reply:
I will try to simulate something if I have time.

Reply:
@jtkech we can update this code (and anywhere we do this gating pattern) with this version which is better: https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/OutputCaching/src/DispatcherExtensions.cs#L8

Usage example: https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/OutputCaching/src/OutputCacheMiddleware.cs#L346

Reply:
@sebastienros Thanks for the info, anyway it looks like that #14859 is due to something else.

So this PR doesn't fix anything but maybe I will merge it because at least it prevents an useless variable closure.

Reply:
> is due to something else

I know, just mentioning it there. Also I'd rather see the lambda use the `x` internally instead of renaming the argument, because now it's ambiguous which one is used if you don't know the precedence rule.

Reply:
Okay no problem I will update it